registry/storage/driver/azure/azure.go

```diff
@@ -282,13 +291,39 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 		return err
 	}
 	destBlobRef := d.client.NewBlockBlobClient(d.blobName(destPath))
-	_, err = destBlobRef.CopyFromURL(ctx, sourceBlobURL, nil)
+	resp, err := destBlobRef.StartCopyFromURL(ctx, sourceBlobURL, nil)
 	if err != nil {
 		if is404(err) {
 			return storagedriver.PathNotFoundError{Path: sourcePath}
 		}
 		return err
 	}
+	copyStatus := *resp.CopyStatus
+	retryCount := 1
+	for copyStatus == blob.CopyStatusTypePending {
+		props, err := destBlobRef.GetProperties(ctx, nil)
+		if err != nil {
+			return err
+		}
+
+		if retryCount >= d.copyStatusPollMaxRetry {
+			destBlobRef.AbortCopyFromURL(ctx, *props.CopyID, nil)
+			return fmt.Errorf("max retries for copy polling reached, aborting copy")
+		}
+
+		copyStatus = *props.CopyStatus
+		if copyStatus == blob.CopyStatusTypeAborted || copyStatus == blob.CopyStatusTypeFailed {
+			if props.CopyStatusDescription != nil {
+				return fmt.Errorf("failed to move blob: %s", *props.CopyStatusDescription)
+			}
+			return fmt.Errorf("failed to move blob with copy id %s", *props.CopyID)
+		}
+
+		if copyStatus == blob.CopyStatusTypePending {
+			time.Sleep(d.copyStatusPollTime)
```

Comment:
Should we do an exponential backoff here? That would help with dropping the default retry count to something smaller than 10

Reply:
gave a stab at it, PTAL